eth_utils/abi.py

```diff
@@ -1,68 +1,784 @@
+    :rtype: `List[str]`
+
+    .. doctest::
+
+        >>> from eth_utils import get_abi_output_types
+        >>> abi = {
+        ...   'constant': False,
+        ...   'inputs': [
+        ...     {
+        ...       'name': 's',
+        ...       'type': 'uint256'
+        ...     }
+        ...   ],
+        ...   'name': 'f',
+        ...   'outputs': [
+        ...     {
+        ...       'name': 'name',
+        ...       'type': 'string'
+        ...     },
+        ...     {
+        ...       'name': 's',
+        ...       'type': 'uint256'
+        ...     }
+        ...   ],
+        ...   'payable': False,
+        ...   'stateMutability': 'nonpayable',
+        ...   'type': 'function'
+        ... }
+        >>> get_abi_output_types(abi)
+        ['string', 'uint256']
+
+    """
+    if abi_element["type"] != "function":
+        raise ValueError(
+            f"Outputs only supported for ABI type `function`. Provided"
+            f" ABI type was `{abi_element.get('type')}` and outputs were "
+            f"`{abi_element.get('outputs')}`."
+        )
+    return [collapse_if_tuple(arg) for arg in abi_element.get("outputs", [])]
 
-def function_abi_to_4byte_selector(function_abi: Dict[str, Any]) -> bytes:
-    function_signature = _abi_to_signature(function_abi)
+
+def function_signature_to_4byte_selector(function_signature: str) -> bytes:
+    r"""
+    Return the 4-byte function selector from a function signature string.
+
+    :param function_signature: String representation of the function name and arguments.
+    :type function_signature: `string`
+    :return: 4-byte function selector.
+    :rtype: `bytes`
+
+    .. doctest::
+
+        >>> from eth_utils import function_signature_to_4byte_selector
+        >>> function_signature_to_4byte_selector('myFunction()')
+        b'\xc3x\n:'
+    """
+    return keccak(text=function_signature.replace(" ", ""))[:4]
+
+
+def function_abi_to_4byte_selector(abi_element: ABIElement) -> bytes:
+    r"""
+    Return the 4-byte function signature of the provided function ABI.
+
+    :param abi_element: ABI element.
+    :type abi_element: `ABIElement`
+    :return: 4-byte function signature.
+    :rtype: `bytes`
+
+    .. doctest::
+
+        >>> from eth_utils import function_abi_to_4byte_selector
+        >>> abi_element = {
+        ...   'type': 'function',
+        ...   'name': 'myFunction',
+        ...   'inputs': [],
+        ...   'outputs': []
+        ... }
+        >>> function_abi_to_4byte_selector(abi_element)
+        b'\xc3x\n:'
+    """
+    function_signature = abi_to_signature(abi_element)
     return function_signature_to_4byte_selector(function_signature)
 
 
 def event_signature_to_log_topic(event_signature: str) -> bytes:
+    r"""
+    Return the 32-byte keccak signature of the log topic for an event signature.
+
+    :param event_signature: String representation of the event name and arguments.
+    :type event_signature: `string`
```

Comment:
nit: we use `str` everywhere else and it's in a code block so let's use `str`

```suggestion
    :type event_signature: `str`
```